src/main/io/gps.c

```diff
@@ -1551,41 +1697,49 @@ static bool UBLOX_parse_gps(void)
         gpsSetFixState(next_fix);
         _new_position = true;
         break;
-    case MSG_STATUS:
+    case MSG_NAV_STATUS:
         *gpsPacketLogChar = LOG_UBLOX_STATUS;
         next_fix = (_buffer.status.fix_status & NAV_STATUS_FIX_VALID) && (_buffer.status.fix_type == FIX_3D);
         if (!next_fix)
             DISABLE_STATE(GPS_FIX);
         break;
-    case MSG_DOP:
-        *gpsPacketLogChar = LOG_UBLOX_DOP;
-        gpsSol.dop.pdop = _buffer.dop.pdop;
-        gpsSol.dop.hdop = _buffer.dop.hdop;
-        gpsSol.dop.vdop = _buffer.dop.vdop;
+    //case MSG_MON_VER:
+    case MSG_NAV_DOP:
+        if (_class == CLASS_MON) {
```

Comment:
The way this reads, we're processing class UBX-NAV-xxx messages. But when we get what appears to be a UBX-NAV-DOP, it's oh wait this is really a class UBX-MON-xxx message so let's do something else instead. We'd have to read the code deeper to notice the variables have 'monVer' in their name to figure out the real message type we're processing.

Some of us might be familiar with the spec, but I think we have to assume some folks will wander into this code who haven't spent hours reading it. I think they could easily get confused here.

Although I'm not a big fan of nested switch statements, in this case I think it might be clearer to have an outer switch for the _class and nested switches for the _msg_id.

`switch (_class) {
case CLASS_MON:
    switch (_msg_id) {
    case MSG_MON_VER:
        ...etc...
    }
case CLASS_NAV:
    switch (_msg_id) {
    case MSG_NAV_POSLLH:
        ...etc., etc...
    }
}`

I think that follows the protocol spec organization of classes & message types, and has a more clear location to find processing for each class & message type, and if we're not handling certain ones, and to add new ones in the future.

Anyway, a suggestion...

Reply:
Another possibility is to combine _class and _msg_id into single integer and switch on that
```
#define CLSMSG(cls, msg) (((cls) << 8) | (msg))
switch (CLSMSG(_class, _msg_id)) {
case CLSMSG(CLASS_NAV, MSG_NAV_DOP):

case CLSMSG(CLASS_MON, MSG_NAV_DOP):
...

} // end case
#undef CLSMSG

Reply:
Closing because I think this is resolved now?